tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/DefaultTaggedMetricRegistry.java

```diff
@@ -22,18 +22,18 @@
 @AutoService(TaggedMetricRegistry.class)
 public final class DefaultTaggedMetricRegistry extends AbstractTaggedMetricRegistry {
 
-    private static final TaggedMetricRegistry DEFAULT = new DefaultTaggedMetricRegistry();
-
     public DefaultTaggedMetricRegistry() {
         super(ExponentiallyDecayingReservoir::new);
```

Comment:
we should probably change this too, feels weird to have a no-args constructor and a no-args "default" static factory that return two different behaviours

Reply:
I actually don't want to change this, because the hierarchy is:

```
TaggedMetricRegistry
├── DefaultTaggedMetricRegistry
└── SlidingWindowTaggedMetricRegistry
```

In `DefaultMetricRegistry.getDefault()`, the 'default' means different things in both cases - the first one could have been 'ExponentiallyDecaying' or 'Impl'. I think the `getDefault()` method could really have been more of a `getSingleton()`.

Basically if we change this constructor then there is no way to get an ExponentiallyDecayingReservoir at all.